lib/machine/configurers/prCloser.ts

```diff
@@ -0,0 +1,133 @@
+import { GoalConfigurer } from "@atomist/sdm-core";
+import { MyGoals } from "../goals";
+import { GetPrsForBranch } from "../../typings/types";
+import { execPromise, slackInfoMessage } from "@atomist/sdm";
+import { logger, TokenCredentials } from "@atomist/automation-client";
+import * as _ from "lodash";
+import * as GithubApi from "@octokit/rest";
+
+export const PrCloserConfigurator: GoalConfigurer<MyGoals> = async (sdm, goals) => {
+  goals.pushImpact.withListener(
+    async i => {
+      /**
+       * Go find all the PRs opened where the destination branch is the branch we are currently operating on
+       */
+      const repos = await i.context.graphClient.query<GetPrsForBranch.Query, GetPrsForBranch.Variables>({
+        name: "GetPrsForBranch",
+        variables: {
+          owner: i.push.repo.owner,
+          repo: i.push.repo.name,
+          branch: i.push.branch,
+        },
+      });
+
+      /**
+       * The Push Impact goal does a shallow clone (depth=2).  We need more history to compare branches
+       */
+      await execPromise("git", ["fetch", "origin", `--depth=1000`], { cwd: i.project.baseDir });
+
+      /**
+       * Find all open PRs targeting this branch
+       */
+      const prsOnThisBranch = _.flatten(repos.Repo[0].branches.map(
+        b => b.pullRequests.filter(pr => pr.state === "open")));
+      const closeThesePrs: Array<{number: number, branch: string}> = [];
```

Comment:
Why don't you move that state filter into the GraphQL query above?

Reply:
Great point...updated